flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/config/entries/ModuleEntry.java

```diff
@@ -61,6 +62,15 @@ private static ModuleEntry create(DescriptorProperties properties) {
         final DescriptorProperties cleanedProperties =
                 properties.withoutKeys(Collections.singletonList(MODULE_NAME));
 
+        // mapping modules purely by name
+        if (cleanedProperties.containsKey(MODULE_TYPE)) {
+            throw new ValidationException(
```

Comment:
I think we should need to keep backward-comaptibility. Using module name as module type is just a shortcut or syntax sugar. 

We can simply fill MODULE_TYPE using module name if it is not defined here.

Reply:
Hi Jark, it seems the summary conclusion missed this part and directly updated the YAML format. 
![image](https://user-images.githubusercontent.com/55568005/107881361-009eab80-6f1f-11eb-9788-28ba98ad0f32.png)

I'll update Jira and make type an optional key in the YAML file to keep backward-compatibility.

Reply:
Or we can close this PR and let YAML unchanged because "it will be removed soon". What do you think?

Reply:
I'm fine with both.